src/portfft/common/global.hpp

```diff
@@ -509,11 +509,12 @@ std::vector<sycl::event> compute_level(
         // level cache.
         cgh.depends_on(dependencies.at(static_cast<std::size_t>(batch_in_l2)));
       }
+      const bool using_wi_level = kd_struct.level == detail::level::WORKITEM;
+      const Scalar* subimpl_twiddles = using_wi_level ? nullptr : twiddles_ptr + subimpl_twiddle_offset;
```

Comment:
Maybe consider adding a comment here to explain that the runtime may check that the pointer is valid even if it is unused?

Reply:
Done in [20ee168](https://github.com/codeplaysoftware/portFFT/pull/125/commits/20ee168aad2472ad45482e3d89a383489cf949db).